File: test_cases/python/claude/13.py
def run_tests(solution_module):
    f = solution_module.numDistinct
    # t not present in s
    assert f('abc', 'z') == 0
    # t equals s — only one way
    assert f('abc', 'abc') == 1
    # empty t — one way (empty subsequence)
    assert f('abc', '') == 1
    # classic example
    assert f('rabbbit', 'rabbit') == 3
    # repeated chars
    assert f('aaa', 'a') == 3

File: test_cases/c/claude/13.c
#include <assert.h>
#include <stdio.h>

int numDistinct(char* s, char* t);

int main(void) {
    assert(numDistinct("rabbbit", "rabbit") == 3);
    assert(numDistinct("babgbag", "bag") == 5);
    assert(numDistinct("a", "a") == 1);
    assert(numDistinct("abc", "abcd") == 0);
    assert(numDistinct("aaaa", "aa") == 6);

    printf("All tests passed.\n");
    return 0;
}